src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+ObstacleParamsParser::ObstacleParamsParser(
+    const boost::shared_ptr<ros::NodeHandle> &pnh,
+    const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+    : pnh_(pnh), collision_model_(collision_model){};
+
+ObstacleParamsParser::~ObstacleParamsParser(){};
+
+void ObstacleParamsParser::addCollisions() {
+  // First adding the obstacles to the geometry model
+
+  std::cout << "in the parser" << std::endl;
+
+  int obs_idx = 1;
+
+  while (pnh_->hasParam("obstacle" + std::to_string(obs_idx) + "/type")) {
+
+    std::string obstacle_name;
+    obstacle_name = "obstacle" + std::to_string(obs_idx);
+
+    std::string type;
+    if (!pnh_->hasParam(obstacle_name + "/type")) {
+      std::cerr << "No obstacle type declared." << std::endl;
+      return;
+    }
+    pnh_->getParam(obstacle_name + "/type", type);
+    std::vector<double> translation_vect;
+    if (!pnh_->hasParam(obstacle_name + "/translation")) {
+      std::cerr << "No obstacle translation declared." << std::endl;
+      return;
+    }
+    pnh_->getParam(obstacle_name + "/translation", translation_vect);
+    const auto translation =
+        Eigen::Map<Eigen::Vector3d>(translation_vect.data());
+
+    std::vector<double> rotation_vect;
+    if (!pnh_->hasParam(obstacle_name + "/rotation")) {
+      std::cerr << "No obstacle rotation declared." << std::endl;
+      return;
+    }
+    pnh_->getParam(obstacle_name + "/rotation", rotation_vect);
+    const auto rotation = Eigen::Map<Eigen::Quaterniond>(rotation_vect.data());
+
+    pinocchio::GeometryObject::CollisionGeometryPtr geometry;
+
+    if (type == "sphere") {
+      double radius;
+      if (pnh_->hasParam(obstacle_name + "/radius")) {
+        pnh_->getParam(obstacle_name + "/radius", radius);
+        geometry = pinocchio::GeometryObject::CollisionGeometryPtr(
+            new hpp::fcl::Sphere(radius));
+      } else {
+        std::cerr << "No dimension or wrong dimensions in the obstacle "
+                     "config. Try to use "
+                     "the ones for the shapes desired, such as 'radius' for "
+                     "'sphere','x', 'y', 'z' for 'box' or 'halfLength' and "
+                     "'radius' for 'capsule'."
+                  << std::endl;
```

Comment:
To avoid repeating this log you can put the whole thing into try-catch block and in case the param is missing raise an exception, which later logs this error. This way you can have the `std::cerr` written only once